### DISCUSSION TITLE
How to setup DNS for subdomain and Emails from root domain

### DISCUSSION THREAD

**Discussion by shanmugad** [modoboa/modoboa#2724] [General]
Hi, 

Modoboa required to add DKIM, SPF, MX, etc record for root domain but how can i set it for subdomain? 

Current record

```
@ MX 10 mail.example.com
Mail @ x.x.x.x (Server IP)
@ TXT v=spf1 include:example.com ~all
```

I want to set it as 

```
email MX 10 mail.example.com
Mail @ x.x.x.x (Server IP)
@ TXT v=spf1 include:email.example.com ~all
```

But my email ID should be `admin@example.com` not `admin@email.example.com`

Expected result : 

![image](https://user-images.githubusercontent.com/121617479/210125774-b44861e6-5b41-412a-b19f-7a75e15f9366.png)

**Comment by Dr-Ari-Gami** [comment]
Let me help you break down these 3 dns records.

**General Info:**
* `@` is an alias for the root domain, in this case, `example.com`.
* When configuring records, treat the name (usually the first field), as if it will automatically be a subdomain, unless you use `@`. For example, setting the name to `email` will actually effect `email.example.com`

**MX Record:** (Points email to the mail server)
* The first field, which is the name, will signify what domain you want to serve email for. In your case, you want it to be your root domain, so use `@`. That way your email address will be `admin@example.com`
* The number is the priority. `10` is fine. This is used when you have backup mail servers.
* The third field is the mail server that serves your email. In your case, you will want to set it to `email.example.com`.

**A Record:** (Points a domain to a server)
* You want to set the name to be `email`. That way you can access your email server at `https://email.example.com`, and the above MX record will have a valid mail server.
* You want to set the IP address to your server's ip address. In your example, `X.X.X.X`.

**SPF Record:** (signifies what servers are allowed to send mail)
* This type of record is a TXT record.
* Set the name to `@`. That will signify that any email received from an `@example.com` address will have to be the following:
* Just set it to `v=spf1 mx ~all`
* That means that any server with an MX record you configured will be authorized. In this case, you are using the same server to both send and receive mail, so that is fine.

The reason that your email can only be sent from `email.example.com` is because you set it that way in your MX record, by having the name be `email` and not `@`. 

Hope this helps!
